src/inmanta/user_setup.py

```diff
@@ -0,0 +1,163 @@
+audience=https://{hostname}:{server_config.server_bind_port.get()}/
+"""
+
+
+def validate_server_setup() -> None:
+    """Validate the server configuration so that authentication is setup correctly."""
+    config.Config.load_config()
+
+    # make sure auth is on
+    if not server_config.server_enable_auth.get():
+        raise click.ClickException(
+            "Server authentication should be enabled before running the initial user setup. "
+            "The option auth in the server section should be enabled."
+        )
+
+    click.echo(f"{'Server authentication: ' : <50}{click.style('enabled', fg='green')}")
+
+    # make sure the method is set to database
+    if server_config.server_auth_method.get() != "database":
+        raise click.ClickException(
+            "The server authentication method should be set to database to continue. Make sure auth_method in the server "
+            "section is set to database"
+        )
+
+    click.echo(f"{'Server authentication method: ' : <50}{click.style('database', fg='green')}")
+
+    # make sure there is auth config that supports signing tokens
+    cfg = config.AuthJWTConfig.get_sign_config()
+    if cfg is None:
+        click.echo("Error: No signing config available in the configuration.")
+
+        value = None
+        while value not in ["yes", "no"]:
+            value = click.prompt("Do you want to generate a new configuration? yes/no")
+
+        if value == "yes":
+            click.echo("Add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+
+        raise click.ClickException("Make sure signing configuration is added to the config. See the documentation for details.")
+
+    click.echo(f"{'Authentication signing config: ' : <50}{click.style('enabled', fg='green')}")
+
+    # TODO: verify web-console config (if any)
+
+
+async def get_database_connection() -> asyncpg.pool.Pool:
+    database_host = server_config.db_host.get()
+    database_port = server_config.db_port.get()
+
+    database_username = server_config.db_username.get()
+    database_password = server_config.db_password.get()
+    connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+    connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+    connection_timeout = server_config.db_connection_timeout.get()
+    return await data.connect(
+        database_host,
+        database_port,
+        server_config.db_name.get(),
+        database_username,
+        database_password,
+        False,
+        connection_pool_min_size=connection_pool_min_size,
+        connection_pool_max_size=connection_pool_max_size,
+        connection_timeout=connection_timeout,
+    )
+
+
+async def do_user_setup() -> None:
+    """Perform the user setup that requires the database interaction"""
+    connection = None
+    try:
+        connection = await get_database_connection()
```

Comment:
Not being able to connect to the database sound like a very plausible issue here. I would:

* Output which database we are trying to connect to. This will make it easier for the user to debug obvious configuration issues.
* I would also make sure there is a clear error message when the connection to the database doesn't work properly. At the moment we will just get a stack trace I guess.